### DISCUSSION TITLE
Blocking Activities Quick Guide

### DISCUSSION THREAD

**Discussion by theafricaninternetgroup** [elsa-workflows/elsa-core#3557] [Q&A] [answered]
I am following along the quick guides on Blocking Activities in the `FileReceivedInvoker` class use is made of a classed called `CollectWorkflowsContext` .In which namespace does this class reside or if this class has been removed in version 2.9.2 how would the new version of the following code snippet be in version 2.9.2
```
public async Task<IEnumerable<CollectedWorkflow>> DispatchWorkflowsAsync(CancellationToken cancellationToken = default)`
        {
            var context = new CollectWorkflowsContext(nameof(FileReceived), new FileReceivedBookmark());
            return await _workflowLaunchpad.CollectAndDispatchWorkflowsAsync(context, null, cancellationToken);
        }
```

**Comment by sfmskywalker** [comment, marked as answer]
Hi @theafricaninternetgroup , the `CollectWorkflowsContext` class has been renamed to `WorkflowsQuery`. I'll update the quick start accordingly. Thanks for reporting!

**Reply by theafricaninternetgroup** [reply]
Thanks
